warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java

```diff
@@ -500,19 +498,38 @@ public ShardIndexQueryTable clone() {
     }
     
     public Boolean getSeparateCountsByColVis() {
-        return separateCountsByColVis;
+        
+        if (config != null) {
```

Comment:
This is looking great.  The only thing I would change is to NOT check if config is null here or in the methods below.  I would rather get a NullPointerException if there is no config as that is a non-starter.  If this logic does not have a config then nothing is going to work well here.

Reply:
Alternatively check for null but throw an IllegalStateException or something like that.